File: 0001-two-sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let lookUP = {}

    for (let i = 0; i < nums.length; i++) {
        if (lookUP[target - nums[i]] == undefined) {
            lookUP[nums[i]] = i;
        }
        else {
            return [lookUP[target - nums[i]], i]
        }
    }
};

// /**
//  * @param {number[]} nums
//  * @param {number} target
//  * @return {number[]}
//  */
// var twoSum = function (nums, target) {
//     for (let i = 0; i < nums.length; i++) {
//         for (let j = i + 1; j < nums.length; j++) {
//             if (nums[i] + nums[j] == target) {
//                 return [i, j]
//             }
//         }
//     }
// };

File: 0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numsMap{};

        for (int i = 0, n = nums.size(); i < n; i++) {
            int comp = target - nums[i];
            if (numsMap.find(comp) != numsMap.end()) {
                return {i, numsMap[comp]};
            } else {
                numsMap[nums[i]] = i;
            }
        }

        return {0, 0};
    }
};